### PR TITLE
Fix config start

### PR DESCRIPTION
**Pull request by sallyom**
* starting with empty config file had a bug, this PR fixes it and adds a workflow to test
* also updates the example unit files to mount the config directory with `/opt/mount` since this allows harpoon to start without a config file and only with the $HARPOON_CONFIG_URL
@cooktheryan @josephsawaya

### CODE REVIEW

**Comment by sallyom** [comment]
for the workflow, since we are not running harpoon with systemd, when starting with no config file, you _do_ need to create the `~/.harpoon` directory because otherwise the `podman run -v $HOME/.harpoon:/opt/mount` will fail. With the unit file, that directory is created with `ExecStartPre`

File: pkg/engine/harpoon.go [hunk 1]
```diff
@@ -145,37 +144,35 @@ func isLocalConfig(v *viper.Viper) (*HarpoonConfig, bool, error) {
...
-				klog.Infof("error processing config file %s, will try to download from HARPOON_CONFIG_URL: %v", hc.configFile, err)
-			}
+	var err error
+	var isLocal, exists bool
+	var config *HarpoonConfig
+	envURL := os.Getenv("HARPOON_CONFIG_URL")
+
+	// user will pass path on local system, but it must be mounted at the defaultConfigPath in harpoon pod
+	// harpoon will read the configFile from within the harpoon pod at /opt/mount/harpoon-config.yaml
+	// harpoon will read the configFile from within the harpoon pod
```
Comment (sallyom) [inline pkg/engine/harpoon.go:154]: will remove the duplicate comment

File: pkg/engine/harpoon.go [hunk 2]
```diff
@@ -184,7 +181,7 @@ func (hc *HarpoonConfig) InitConfig(initial bool) {
 		}
 	}
 
-	if config.Targets == nil {
+	if config != nil && config.Targets == nil {
 		cobra.CheckErr("no harpoon targets found, exiting")
```
Comment (sallyom) [inline pkg/engine/harpoon.go:185]: will update this to if config == nil || config.Targets == nil

File: pkg/engine/helpers.go [hunk 3]
```diff
@@ -35,10 +35,21 @@ func downloadUpdateConfigFile(urlStr string, existsAlready bool) (bool, error) {
...
+		return false, fmt.Errorf("found empty config at %s, unable to update config", urlStr)
+	}
+	// if initial, this is the last resort, newBytes should be populated
+	// the only way to get here from initial
+	// is if there is no config file on disk, only a HARPOON_CONFIG_URL
+	if initial {
+		if err := os.WriteFile(defaultConfigPath, newBytes, 0600); err != nil {
+			return false, fmt.Errorf("could not copy contents of %s to path %s: %v", urlStr, defaultConfigPath, err)
+		}
+		return true, nil
```
Comment (sallyom) [inline pkg/engine/helpers.go:50]: this is the fix, without this, the contents of the config url were never placed in config file, so harpoon failed - in testing b4, I must've had a stale config file at ~/.harpoon/config.yaml so didn't catch this. I've added a workflow for it now.

File: pkg/engine/harpoon.go (hunk 2 above)
Comment (sallyom) [reply pkg/engine/harpoon.go:185]: done

**Review by cooktheryan — APPROVED** [review]

**Pull request merged**